File: UVa/Problem Solving/Greedy/UVa_410_Station Balance.py
# https://onlinejudge.org/external/4/410.pdf

def solution(c, s, masses):
    avg = sum(masses) / c
    imbalance = 0

    if s < 2 * c:
        masses += [0] * (2 * c - s)
    masses.sort()

    for i in range(c):
        imbalance += abs(masses[i] + masses[-1-i] - avg)
        print(f"{i:2}:", end="")
        if masses[i]:
            print(f" {masses[i]}", end="")
        if masses[-1-i]:
            print(f" {masses[-1-i]}", end="")
        print()
    print(f"IMBALANCE = {imbalance:.5f}\n")


count = 1
while True:
    try:
        c, s = map(int, input().split())
        masses = list(map(int, input().split()))
        print(f"Set #{count + 1}")
        solution(c, s, masses)
        count += 1
    except EOFError:
        break

File: UVa/Problem Solving/Greedy/UVa_410_Station Balance.cpp
// https://onlinejudge.org/external/4/410.pdf

#include <bits/stdc++.h>
using namespace std;
#define REP(i, n) for (int i = 0; i < n; ++i)
#define RESET(a, b) memset(a, b, sizeof(a))

int C, S;
int specimens[10];

int main()
{
    int t = 1;
    while (scanf("%d %d", &C, &S) != EOF)
    {
        printf("Set #%d\n", t++);
        RESET(specimens, 0);

        double avg = 0, imbalance = 0;
        REP(i, S)
        {
            scanf("%d", &specimens[i]);
            avg += specimens[i];
        }
        avg /= C;
        sort(specimens, specimens + 2 * C);
        REP(i, C)
        {
            imbalance += abs((specimens[i] + specimens[2 * C - 1 - i]) - avg);
            printf(" %d:", i);
            if (specimens[i])
                printf(" %d", specimens[i]);
            if (specimens[2 * C - 1 - i])
                printf(" %d", specimens[2 * C - 1 - i]);
            puts("");
        }
        printf("IMBALANCE = %.5f\n\n", imbalance);
    }

    return 0;
}